pkg/repo/callrepo.go

```diff
@@ -93,6 +101,23 @@ func (r *CallsRepository) GetCall(method string) core.Handler {
 	return r.calls[method]
 }
 
+// UpdateCalls refreshes the CallsConfig and rebuilds the handlers for calls accordingly
+// It takes a single parameter that is pointer to the new calls config.
+// The current implementation will completely overwrite the old config with new config.
+// It returns an error to if any while building the handlers, otherwise it returns nil.
+func (r *CallsRepository) UpdateCalls(calls *CallsConfig) error {
+	r.calls = make(map[string]core.Handler)
+
+	for _, call := range calls.Calls {
+		err := createHandler(call, r)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
this should be atomic operation and protected with mutex, we should create separate map with handlers, as you're doing. but you should not assign it to repo straight away. when map is ready, no errors identify, get lock and replace map. 

with current implementation calls will be failing while we're updating the map, if for some reason we'll not be able to update map it'll left server in inconsistent state

Reply:
Agree. Actually all etcd handling should be mutually exclusive.